machine/v1beta1/types_alibabaprovider.go

```diff
@@ -0,0 +1,329 @@
+	// Valid values:
+	//PL0: A single ESSD can deliver up to 10,000 random read/write IOPS.
+	//PL1: A single ESSD can deliver up to 50,000 random read/write IOPS.
+	//PL2: A single ESSD can deliver up to 100,000 random read/write IOPS.
+	//PL3: A single ESSD can deliver up to 1,000,000 random read/write IOPS.
+	//For more information about ESSD performance levels, see ESSDs.
+	// +optional
+	PerformanceLevel string `json:"performanceLevel,omitempty"`
+
+	// Name is the name of the system disk. The name must be 2 to 128 characters in length. It must start with a letter and cannot start with http:// or https://. It can contain letters, digits, colons (:), underscores (_), and hyphens (-).
+	//This parameter is empty by default.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// Size is the size of the system disk. Unit: GiB. Valid values: 20 to 500.
+	// The value must be at least 20 and greater than or equal to the size of the image.
+	// The default value is 40 or the size of the image, depending on whichever is greater.
+	// +optional
+	Size int `json:"size,omitempty"`
+
+	// Description is the description of the system disk. The description must be 2 to 256 characters in length and cannot start with http:// or https://.
+	// This parameter is empty by default.
+	// +optional
+	Description string `json:"description,omitempty"`
+}
+
+// DataDisk contains the information regarding the datadisk attached to an instance
+type DataDiskProperties struct {
+	// Name is the name of data disk N. The name must be 2 to 128 characters in length. It must start with a letter and cannot start with http:// or https://. It can contain letters, digits, colons (:), underscores (_), and hyphens (-).
+	//
+	// This parameter is empty by default.
+	// +optional
+	Name string `name:"diskName,omitempty"`
+
+	// SnapshotID is the ID of the snapshot used to create data disk N. Valid values of N: 1 to 16.
+	//
+	// When the DataDisk.N.SnapshotID parameter is specified, the DataDisk.N.Size parameter is ignored. The data disk is created based on the size of the specified snapshot.
+	// Use snapshots created after July 15, 2013. Otherwise, an error is returned and your request is rejected.
+	// +optional
+	SnapshotID string `name:"snapshotId,omitempty"`
+
+	// Size of the data disk N. Valid values of N: 1 to 16. Unit: GiB. Valid values:
+	//
+	// Valid values when DataDisk.N.Category is set to cloud_efficiency: 20 to 32768
+	// Valid values when DataDisk.N.Category is set to cloud_ssd: 20 to 32768
+	// Valid values when DataDisk.N.Category is set to cloud_essd: 20 to 32768
+	// Valid values when DataDisk.N.Category is set to cloud: 5 to 2000
+	// The value of this parameter must be greater than or equal to the size of the snapshot specified by the SnapshotID parameter.
+	// +optional
+	Size int `name:"size,omitempty"`
+
+	// DiskEncryption specifies whether to encrypt data disk N.
+	//
+	//Default value: disabled.
+	// +optional
```

Comment:
we should add the kubebuilder annotation for the enum validation here, as described in the [kubebuilder docs](https://book.kubebuilder.io/reference/markers/crd-validation.html#crd-validation)

i think it should look like this
```suggestion
        // +kubebuilder:validation:Enum="encrypted";"disabled"
	// +optional
```